### DISCUSSION TITLE
Liquidus Temperature

### DISCUSSION THREAD

**Discussion by AadhiK2012** [pycalphad/pycalphad#395] [Q&A] [answered]
Does pycalphad has an in-built function that takes in alloy composition as input and returns the liquidus Temperature? 
A search over temperature using equilibrium function works but wanted to know about the availability of an in-built function.

Thanks!

**Comment by bocklund** [comment, marked as answer]
There's not currently anything built-in and exposed at a high level. With pycalphad 0.9 and later, it's possible to use the internal API to add a `CompositionSet` with fixed amount (e.g. liquid with a fixed amount of 1), but this is an advanced operation currently.

In the `scheil` package, we have some code to do a binary search for the solidus that could be adapted to computing the liquidus. Here's the snippet ([link](https://github.com/pycalphad/scheil/blob/68183921c58310fd8e8451807a01225dd8349246/scheil/simulate.py#L320-L340)):

```python
            # binary search to find the solidus
            T_high = current_T + step_temperature  # High temperature, liquid
            T_low = current_T  # Low temperature, solids only
            found_ph = set(eq.Phase[eq.Phase != ''].tolist())
            if verbose:
                print(f'Found phases {found_ph}. Starting binary search between T={(T_low, T_high)} ', end='')
            while (T_high - T_low) > binary_search_tol:
                bin_search_T = (T_high - T_low) * 0.5 + T_low
                conds[v.T] = bin_search_T
                eq = equilibrium(dbf, comps, phases, conds, model=models, phase_records=phase_records, to_xarray=False, **eq_kwargs)
                if adaptive:
                    # Update the points dictionary with local samples around the equilibrium site fractions
                    _update_points(eq, eq_kwargs['calc_opts']['points'], dof_dict)
                if not is_converged(eq):
                    if verbose:
                        comp_conds = {cond: val for cond, val in conds.items() if isinstance(cond, v.X)}
                        print(f"Convergence failure at T={conds[v.T]} X={comp_conds} ")
                if liquid_phase_name in eq.Phase:
                    T_high = bin_search_T
                else:
                    T_low = bin_search_T
```

The `equilibrium` arguments `model=models, phase_records=phase_records, to_xarray=False` are performance related and you don't have to include themto get a simple example up and running

**Reply by AadhiK2012** [reply]
Thank you for the reply. This will work

**Comment by xiaowei10** [comment]
Can we use the temperature of the liquidus of a binary system as the melting point of the corresponding composition?

**Reply by richardotis** [reply]
The liquidus temperature is the temperature at which, when cooling (and assuming equilibrium), the first solid begins to form. The solidus temperature is the temperature at which, when cooling, the last liquid disappears. The difference between the two is called the freezing range. It may make sense to use the liquidus temperature as a 'melting temperature' for an alloy, but it depends on your application. If you're looking at stability of an alloy for a high-temperature application, you probably care about the solidus more than the liquidus.

**Comment by xiaowei10** [comment]
Can we get the melting point of a  ternary compound via pyCalphad?
